Makefile

```diff
@@ -54,20 +56,17 @@ install: build
 	$(DKMS) add hid-xpadneo
 
 uninstall: VERSION
-	$(DKMS) remove "hid-xpadneo/$(shell cat VERSION)" --all
+	$(DKMS) remove "hid-xpadneo/$(shell cat VERSION)" --all || true
 	rm -Rf "$(PREFIX)/usr/src/hid-xpadneo-$(shell cat VERSION)"
 	rm -f $(DOCS:%=$(PREFIX)$(DOC_PREFIX)/%)
 	rm -f $(UDEV_RULES:%=$(PREFIX)$(ETC_PREFIX)/udev/rules.d/%)
 	rm -f $(MODPROBE_CONFS:%=$(PREFIX)$(ETC_PREFIX)/modprobe.d/%)
 	rmdir --ignore-fail-on-non-empty -p $(PREFIX)$(ETC_PREFIX)/modprobe.d $(PREFIX)$(ETC_PREFIX)/udev/rules.d $(PREFIX)$(DOC_PREFIX)
 
 sign:
-ifeq ($(NO_SKIP_SIGN), y)
-	@openssl req -new -x509 -newkey rsa:2048 -keyout MOK.priv -outform DER -out MOK.der -nodes -days 36500 -subj "/CN=Custom MOK/"
-	@mokutil --import MOK.der
-else
-	echo "Skipping key creation"
-endif
-	@$(KSRC)/scripts/sign-file sha256 MOK.priv MOK.der hid-xpadneo.ko
+	@sudo update-secureboot-policy --new-key
```

Comment:
The Makefile should not call sudo, especially it should not do so hidden, and it should not run by surprise (e.g., you may have unlocked sudo before, and then it would run without asking for passwords). This also makes no sense because `modprobe` requires sudo but doesn't use it here. I'd suggest to rather add a guard `[ "$(id -u)" = "0"  ] || { echo >&2 "need to run with sudo"; exit 1; }`.

There are exceptions to this rule but this one is obviously not it.